integration/mounts/mounts.go

```diff
@@ -0,0 +1,193 @@
+package mounts
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	socketPath     = "/var/run/vc/vm/foobar/console.sock"
+	kataConfigPath = "/usr/share/kata-containers/configuration.toml"
+)
+
+var (
+	kataConfigBackup = strings.Join([]string{kataConfigPath, "bak"}, ".")
+)
+
+func testOverConsole(cmd, match, socket string) bool {
+	conn, err := net.DialTimeout("unix", socket, 10*time.Second)
+	if err != nil {
+		fmt.Println("failed to connect: %w", err)
+		return false
+	}
+	defer conn.Close()
+
+	if _, err = conn.Write([]byte(cmd)); err != nil {
+		conn.Close()
+		return false
+	}
+
+	errChan := make(chan error)
+
+	go func() {
+		reader := bufio.NewReader(conn)
+		// expect echo from shell, skip what we sent and ignore first response
+		if _, err := reader.ReadString('\n'); err != nil {
+			errChan <- err
+			fmt.Println("failed to read: ", err)
+			return
+		}
+
+		if _, err = reader.ReadString('\n'); err != nil {
+			errChan <- err
+			fmt.Println("failed to read: ", err)
+			return
+		}
+
+		for {
+			response, err := reader.ReadString('\n')
+			if err != nil {
+				errChan <- err
+				fmt.Println("failed to read: ", err)
+				return
+			}
```

Comment:
These three identical code blocks are rather confusing to me I'm afraid. Could you simplify and add more comment detail?